js/scripts.js

```diff
@@ -1,33 +1,85 @@
 var form_login = document.getElementById('login-form');
 
-
-form_login.onsubmit=function(e)
-{
+form_login.onsubmit = function (e) {
     e.preventDefault();
+    $("#checkbox").on('change', function () {
+        if ($(this).is(':checked')) {
+            $(this).attr('value', 'true');
+        } else {
+            $(this).attr('value', 'false');
+        }
 
-    var email=form_login.email.value;
-    var pass=form_login.pass.value;
+        $('#checkbox-value').text($('#checkbox').val());
+    });
+    var email = form_login.email.value;
+    var pass = form_login.pass.value;
+    var check = form_login.chkRem.value;
+    var passhash = CryptoJS.MD5(pass).toString();
+    var remember = false;
 
     console.log(email);
-    console.log(pass);
+    console.log(passhash);
+    console.log(check);
+
+    if (check) {
+        setCookie("username", email, 1);
+    }
 
     $.ajax({
         type: "POST",
-        url: "",
+        url: "https://soy-sauce.herokuapp.com/user/login",
         data:
             {
-                username:email,
-                password:pass
+                email: email,
+                password: passhash,
+                remember_me: remember
             },
-        success: function(msg)
-        {
+        success: function (msg) {
             console.log(msg);
         },
-        fail: function()
-        {
+        fail: function () {
             console.log("login failed");
         }
     });
 
 
 }
+window.onload = checkCookie();
+
+function setCookie(cname, cvalue, exdays) {
+    var d = new Date();
+    d.setTime(d.getTime() + (exdays * 24 * 60 * 60 * 1000));
+    var expires = "expires=" + d.toGMTString();
+    document.cookie = cname + "=" + cvalue + "," + expires;
```

Comment:
I doubt whether you should set cookie like this. As far as I know, when the log in is successful, backend will return a response with "Set-Cookie" in the headers, whose value is the actual cookie you should set, and the value is encrypted.

I guess there are two possibilities for this cookie thing:

1. you should set cookie after logging in, but the value should be read from "Set-Cookie" in the header
2. you don't need to set cookie manually at all

I'm not sure which is the case here; we can leave this in the next PR.

Reply:
In my opinion, once user submit the form, I will check the checkbox status and set cookie if its status is checked instead of waiting a response from backend. I put the cookie in the local storage.

Reply:
The thing is it's not a good idea to store un-encryped key-values in cookies, because cookies are meant for the backend to read/write. Otherwise, some malicious user can set the username in his/her cookie to be someone else's username using the Chrome dev tool, which is a security issue.

Reply:
I didn't see any "localstorage" thing in your code, can you please show me where it is?

Reply:
Sorry, I made a mistake. Please ignore the last sentence I replied. I do send the cookie to the backend but I set cookies immediately after user submit the form.

Reply:
I hashed the password by md5 (I will change the method to base64 later). I store username instead of un-encryped key-value in cookie and give it to backend.

Reply:
Here, let's talk about cookie in an easy way.

When the first request happens, front-end don't need to set cookie itself, also there is no any cookie sent to back-end.

Once back-end send a response, there will be an cookie inside of the response.

In the second request from front-end, browser will handle the cookie automatically, so I think you don't need to worry about it.

Reply:
Sure you can use md5 and base64, but this process of encryption are all visible to users, and they can still pretend to be another user by using the same encryption method written in your js file, because the backend has no idea if the new cookie sent back to it is modified/legit or not. Therefore, for the backend (and the whole website), in order to make sure the values are not modified by a malicious user, encrypting cookies can only be done in the backend using a method that no frontend user knows.